internal/controllers/parser.go

```diff
@@ -479,6 +484,42 @@ func UpdateConfigFromAPIOpts(
 		}
 	}
 
+	if opts.Security != nil && opts.Security.Crunch42 != nil {
+		secret, err := k8sutils.GetSecret(context.Background(), kubernetesClient, opts.Security.Crunch42.Token.Name, opts.Security.Crunch42.Token.Namespace)
+		if err != nil {
+			return err
+		}
+		fmt.Println(secret)
+
+		clientSecret := string(secret.Data["client_secret"])
+
+		crunchClient, err := crunch.NewClient(clientSecret, nil)
+		if err != nil {
+			return err
+		}
+		coll, _, err := crunchClient.CreateCollection(&crunch.Collection{Name: name})
+		if err != nil {
+			return err
+		}
+
+		jsn, _ := json.Marshal(spec)
```

Comment:
Missing error checking?

Reply:
it was intentional, but I will add the check regardless